### ISSUE TITLE
Impact of file name changes on links

### ISSUE THREAD

**Issue by brownsarahm** [carpentries/instructor-training#1219]
The upcoming revision will include significant reordering of content and renaming of episode files, so current links to material in this report will get broken.  

Could we add, in the revision, a rendered table (that could live under extras) that helps people find what they were looking for? 

And link to that on this repo's 404?

This would also be useful for when we reorient trainers.

**Comment by Denubis** [comment]
You're proposing a for-loop in the 404 template that is the syllabus page?

**Comment by Denubis** [comment]
Because it just came up for me and I was deeply puzzled -- the lesson won't render if there are broken internal links. To test this it must be run locally with jekyll server until all the internal links resolve.

**Comment by brownsarahm** [comment]
we'd fix the internal links; but external links (eg that people have saved, blog posts, etc) will be broken. 

I'm proposing adding something like the following to the 404

```
We recently reorganized this lesson, a guide to what you may have been looking for is located on the [chages page](extras/changetable)
```

**Comment by Denubis** [comment]
Nice. And while we can link to the syllabus, that'll be renamed and whatnot, so just linking that is insufficient. I support this plan. Trying to maintain historical links strikes me as something that'll be a maintenance nightmare.

**Comment by karenword** [comment]
Good idea! @fmichonneau or @zkamvar can you weigh in on feasibility for this re modifying the 404 message?

**Comment by fmichonneau** [comment]
We'll discuss this during our curriculum team meeting tomorrow and we'll get back to you. Thanks!

**Comment by zkamvar** [comment]
From the outside, I struggle to find context, but here's what I can piece together:

This issue stems from #1217, which will be fixed by #1215, but that PR doesn't currently contain any content. Can someone point me to a place where I can see what the revisions will be?

**Comment by brownsarahm** [comment]
it's a major reivsion that is not done yet.  episodes will be renamed, and coverage of topics will be moved from one episode to another. this summary table will be useful even in the latter case that doesn't address the broken links. 

there are some piece of content in [data lessons](https://github.com/data-lessons/instructor-training/pulls)

**Comment by zkamvar** [comment]
Thank you for the context!

**Comment by brownsarahm** [comment]
Noting here that there have been recent improvements to the 404 page in styles that we should consider in how we handle this: https://github.com/carpentries/styles/pull/578

**Comment by brownsarahm** [comment]
Now that the major update is merged, maybe we should return to this? 

For example, some episodes have changed order and been renamed.  

Everything internal is fine, but if someone was linking to 21-carpentries from somewhere else, that's now 15-carpentries. They'll get a 404 that doesn't help them find the information.

**Comment by fmichonneau** [comment]
for pages that have a direct mapping (like 21-carpentries to 15-carpentries), we could set up redirects, for the others, I'm not sure what else we could do

**Comment by Denubis** [comment]
There's a brute force solution here, which is to generate 301 redirect pages for all the old pages. Instead, would it be worth catching 404s https://jekyllrb.com/tutorials/custom-404-page/ and provide a custom 404 with the episode list in?
